--- a/hadrian/src/CommandLine.hs
+++ b/hadrian/src/CommandLine.hs
@@ -1,5 +1,5 @@
 module CommandLine (
-    optDescrs, cmdLineArgsMap, cmdFlavour, lookupFreeze1, lookupFreeze2, lookupSkipDepends,
+    optDescrs, cmdLineArgsMap, cmdFlavour, lookupFreeze1, lookupFreeze2, lookupFreezeLibs, lookupSkipDepends,
     cmdBignum, cmdBignumCheck, cmdProgressInfo, cmdCompleteSetting,
     cmdDocsArgs, lookupBuildRoot, TestArgs(..), TestSpeed(..), defaultTestArgs,
     cmdPrefix, DocArgs(..), defaultDocArgs
@@ -27,6 +27,7 @@ data CommandLineArgs = CommandLineArgs
     , flavour        :: Maybe String
     , freeze1        :: Bool
     , freeze2        :: Bool
+    , freezeLibs     :: Bool
     , skipDepends    :: Bool
     , bignum         :: Maybe String
     , bignumCheck    :: Bool
@@ -46,6 +47,7 @@ defaultCommandLineArgs = CommandLineArgs
     , flavour        = Nothing
     , freeze1        = False
     , freeze2        = False
+    , freezeLibs     = False
     , skipDepends    = False
     , bignum         = Nothing
     , bignumCheck    = False
@@ -139,6 +141,9 @@ readBuildRoot ms =
 readFreeze1, readFreeze2, readSkipDepends :: Either String (CommandLineArgs -> CommandLineArgs)
 readFreeze1 = Right $ \flags -> flags { freeze1 = True }
 readFreeze2 = Right $ \flags -> flags { freeze1 = True, freeze2 = True }
+
+readFreezeLibs :: Either String (CommandLineArgs -> CommandLineArgs)
+readFreezeLibs = Right $ \flags -> flags { freeze1 = True, freezeLibs = True }
 readSkipDepends = Right $ \flags -> flags { skipDepends = True }

 readProgressInfo :: Maybe String -> Either String (CommandLineArgs -> CommandLineArgs)
@@ -289,6 +294,8 @@ optDescrs =
       "Freeze Stage1 GHC."
     , Option [] ["freeze2"] (NoArg readFreeze2)
       "Freeze Stage2 GHC."
+    , Option [] ["freeze-libs"] (NoArg readFreezeLibs)
+      "Do not rebuild Stage2 libraries but allow executables to be relinked."
     , Option [] ["skip-depends"] (NoArg readSkipDepends)
       "Skip rebuilding dependency information."
     , Option [] ["bignum"] (OptArg readBignum "BACKEND")
@@ -397,6 +404,9 @@ lookupFreeze1 = freeze1 . lookupExtra defaultCommandLineArgs
 lookupFreeze2 :: Map.HashMap TypeRep Dynamic -> Bool
 lookupFreeze2 = freeze2 . lookupExtra defaultCommandLineArgs

+lookupFreezeLibs :: Map.HashMap TypeRep Dynamic -> Bool
+lookupFreezeLibs = freezeLibs . lookupExtra defaultCommandLineArgs
+
 lookupSkipDepends :: Map.HashMap TypeRep Dynamic -> Bool
 lookupSkipDepends = skipDepends . lookupExtra defaultCommandLineArgs

--- a/hadrian/src/Main.hs
+++ b/hadrian/src/Main.hs
@@ -41,11 +41,18 @@ main = do

         rebuild = [ (RebuildLater, buildRoot -/- "stage0/**")
                   | CommandLine.lookupFreeze1 argsMap ||
-                    CommandLine.lookupFreeze2 argsMap
+                    CommandLine.lookupFreeze2 argsMap ||
+                    CommandLine.lookupFreezeLibs argsMap
                   ] ++
                   [ (RebuildLater, buildRoot -/- "stage1/**")
                   | CommandLine.lookupFreeze2 argsMap
                   ] ++
+                  (if CommandLine.lookupFreezeLibs argsMap
+                   then [ (RebuildLater, buildRoot -/- "stage1/libraries/**")
+                        , (RebuildLater, buildRoot -/- "stage1/compiler/**")
+                        , (RebuildLater, buildRoot -/- "stage1/rts/**")
+                        ]
+                   else []) ++
                   (if CommandLine.lookupSkipDepends argsMap
                    then [(RebuildLater, buildRoot -/- "**/.dependencies.mk"), (RebuildLater, buildRoot -/- "**/.dependencies")]
                    else [])
