--- a/rts/StgRun.h
+++ b/rts/StgRun.h
@@ -7,7 +7,11 @@
  *
  * ---------------------------------------------------------------------------*/

-RTS_PRIVATE StgRegTable * StgRun (StgFunPtr f, StgRegTable *basereg);
+#if defined(powerpc64le_HOST_ARCH)
+StgRegTable * StgRun (StgFunPtr f, StgRegTable *basereg);
+#else
+RTS_PRIVATE StgRegTable * StgRun (StgFunPtr f, StgRegTable *basereg);
+#endif

 #if defined(USE_MINIINTERPRETER)
 RTS_PRIVATE StgWord stg_stop_thread_info;
