--- /home/memento/PycharmProjects/Conda-Feedstocks/ghc-feedstock-9.10.2/build_artifacts/src_cache/ghc-9_10_3-src_d266864b/libraries/process/configure	2025-09-10 11:06:02.000000000 -0500
+++ ghc-9.10.3-pristine/libraries/process/configure	2025-11-30 09:54:16.667691608 -0600
@@ -3981,7 +3981,11 @@
 if eval \${$as_fp_Cache+:} false; then :
   $as_echo_n "(cached) " >&6
 else
-  if ac_fn_c_compute_int "$LINENO" "$fp_const_name" "fp_check_const_result"        "$ac_includes_default"; then :
+  if ac_fn_c_compute_int "$LINENO" "$fp_const_name" "fp_check_const_result"        "$ac_includes_default
+#ifdef HAVE_SIGNAL_H
+# include <signal.h>
+#endif
+"; then :
 
 else
   fp_check_const_result='-1'
