--- rts/configure
+++ rts/configure
@@ -3963,6 +3963,9 @@

       # As far as I'm aware, none of these have relevant variants
+      osx)
+        BuildOS="darwin"
+        ;;
       freebsd|dragonfly|hpux|linuxaout|kfreebsdgnu|freebsd2|darwin|nextstep2|nextstep3|sunos4|ultrix|haiku)
         BuildOS="$1"
         ;;
@@ -4284,7 +4287,7 @@
   esac
         ;;
-        darwin|ios|watchos|tvos)
+        darwin|osx|ios|watchos|tvos)
             test -z "$2" || eval "$2=OSDarwin"
             test -z "$3" || eval "$3=ArchUnknown"
             ;;
@@ -4481,6 +4484,9 @@

       # As far as I'm aware, none of these have relevant variants
+      osx)
+        HostOS="darwin"
+        ;;
       freebsd|dragonfly|hpux|linuxaout|kfreebsdgnu|freebsd2|darwin|nextstep2|nextstep3|sunos4|ultrix|haiku)
         HostOS="$1"
         ;;
@@ -7847,7 +7853,7 @@
         ;;
-    darwin|ios|watchos|tvos)
+    darwin|osx|ios|watchos|tvos)
         RtsLinkerUseMmap=1
         ;;
     *)
