--- a/hadrian/src/Settings/Default.hs
+++ b/hadrian/src/Settings/Default.hs
@@ -90,7 +90,7 @@ defaultLibraryWays = Set.fromList <$>
     mconcat
     [ pure [vanilla]
     , notStage0 ? pure [profiling]
-    , notStage0 ? platformSupportsSharedLibs ? pure [dynamic, profilingDynamic]
+    , notStage0 ? platformSupportsSharedLibs ? pure [dynamic]
     ]

 -- | Default build ways for the RTS.
@@ -102,8 +102,6 @@ defaultRtsWays = Set.fromList <$>
     , notStage0 ? targetSupportsThreadedRts ? pure [threaded, threadedProfiling, threadedDebug]
     , notStage0 ? platformSupportsSharedLibs ? pure [dynamic, debugDynamic]
     , notStage0 ? platformSupportsSharedLibs ? targetSupportsThreadedRts ? pure [threadedDynamic, threadedDebugDynamic]
-    , notStage0 ? platformSupportsSharedLibs ? pure [profilingDynamic, debugProfilingDynamic]
-    , notStage0 ? platformSupportsSharedLibs ? targetSupportsThreadedRts ? pure [threadedProfilingDynamic]
     ]

 -- | Default build flavour. Other build flavours are defined in modules
