--- a/configure
+++ b/configure
@@ -7642,7 +7642,7 @@
   if test $? -eq 0 ; then
     # We trim \n due of difference between Emscripten CC and clang.
     # Clang version preserves whitespace lines, but emcc removes.
-    if test "$(cat ./conftest.pp.js | tr -d '\n')" = "// 1" ; then
+    if test "$(cat ./conftest.pp.js | tr -d '\r\n')" = "// 1" ; then
       { $as_echo "$as_me:${as_lineno-$LINENO}: result: yes" >&5
 $as_echo "yes" >&6; }
     else
