--- a/hadrian/src/Builder.hs
+++ b/hadrian/src/Builder.hs
@@ -247,7 +247,8 @@
             return $ [ unlitPath ]
                   ++ ghcdeps
                   ++ [ touchyPath          | windowsHost ]
-                  ++ [ root -/- mingwStamp | windowsHost ]
+                  -- Skip mingw bundling - conda-forge provides toolchain
+                  -- ++ [ root -/- mingwStamp | windowsHost ]
                      -- proxy for the entire mingw toolchain that
                      -- we have in inplace/mingw initially, and then at
                      -- root -/- mingw.
